caf-audit-binding-elasticsearch/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java

```diff
@@ -36,37 +31,54 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Arrays;
+import org.elasticsearch.client.indices.IndexTemplatesExistRequest;
+import org.elasticsearch.client.indices.GetIndexTemplatesResponse;
 
 public class ElasticAuditIndexManager {
 
     private static final Logger LOG = LogManager.getLogger(ElasticAuditIndexManager.class.getName());
+    private static final String INDEX_TEMPLATE_NAME = "caf_audit_template";
 
     private final RestHighLevelClient restHighLevelClient;
-    private final LoadingCache<String, String> indexCache;
 
     private final int numberOfShards;
     private final int numberOfReplicas;
 
-    private XContentBuilder cafAuditEventTenantIndexMappingsBuilder;
-
     public ElasticAuditIndexManager(final int numberOfShards, final int numberOfReplicas, RestHighLevelClient restHighLevelClient) {
         this.restHighLevelClient = restHighLevelClient;
         this.numberOfShards = numberOfShards;
         this.numberOfReplicas = numberOfReplicas;
-
-        //  Configure in memory cache to hold list of Elasticsearch indexes already created.
-        indexCache = CacheBuilder
-                .newBuilder()
-                .build(
-                        //  CacheLoader used to create new index and load into cache if not already cached.
-                        new CacheLoader<String, String>() {
-                            @Override
-                            public String load(String indexName) {
-                                createIndex(indexName);
-                                return indexName;
-                            }
-                        }
-                );
+    }
+    public void createIndexTemplate()
+    {
+        try {
+            if (isIndexTemplatePresent()) {
+                return;
+            }
+            final PutIndexTemplateRequest request = new PutIndexTemplateRequest(INDEX_TEMPLATE_NAME);
+            //  Configure the number of shards and replicas the new index should have.
+            final Settings indexSettings = Settings.builder()
+                .put("number_of_shards", numberOfShards)
+                .put("number_of_replicas", numberOfReplicas)
+                .build();
+            request.settings(indexSettings);
+            request.mapping(getTenantIndexTypeMappingsBuilder());
+            request.patterns(Arrays.asList("*" + ElasticAuditConstants.Index.SUFFIX));
+            restHighLevelClient.indices().putTemplate(request, RequestOptions.DEFAULT);
+        } catch (final IOException ex) {
+            LOG.error("An error occured contacting elasticsearch: ", ex);
```

Comment:
The IOException is only logged, meaning the connection would appear to be valid.